features/netsocket/mbed_ipstack.h

```diff
@@ -0,0 +1,123 @@
+/* mbed OS IP stack API
+ * Copyright (c) 2015-2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_IPSTACK_H
+#define MBED_IPSTACK_H
+
+#include "nsapi.h"
+#include "emac_api.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * mbed OS API for IP stack abstraction
+ *
+ * This interface should be used by targets to initialize IP stack, create, bring up and bring down network interfaces.
+ */
+
+/** Initialize IP stack
+ *
+ * This function should be called before any of the network interfaces is added. It's up to target's setup code, to make
+ * sure that the stack is initialized and all the existing interfaces are registered with the stack.
+ * This function can be safely called multiple times, it will do nothing and return NSAPI_ERROR_OK if stack is already
+ * initialized.
+ *
+ * @return    NSAPI_ERROR_OK on success, or error code
+ */
+void mbed_ipstack_init(void);
+
+/** Register a network interface with the IP stack
+ *
+ * Connects EMAC layer with the IP stack and initializes all the required infrastructure.
+ * This function should be called only once for each available interface.
+ *
+ * @param    emac       EMAC HAL implementation for this network interface
+ * @param    default_if true if the interface should be treated as the default one
+ * @return              NSAPI_ERROR_OK on success, or error code
+ */
+nsapi_error_t mbed_ipstack_add_netif(emac_interface_t *emac, bool default_if);
```

Comment:
Should this be "add_netif" still? netif is the lwip term, maybe "add_interface"?